st2common/st2common/models/db/policy.py

```diff
@@ -0,0 +1,180 @@
+        if not resource_type or not name:
+            raise ValueError('Both resource_type and name are required for building ref. '
+                             'resource_type=%s, name=%s' % (resource_type, name))
+
+        resource_type = cls.validate_resource_type(resource_type)
+        return cls.separator.join([resource_type, name])
+
+    @classmethod
+    def validate_resource_type(cls, resource_type):
+        if not resource_type:
+            raise ValueError('Resource type should not be empty.')
+
+        if cls.separator in resource_type:
+            raise ValueError('Resource type should not contain "%s".' % cls.separator)
+
+        return resource_type
+
+    @classmethod
+    def get_resource_type(cls, ref):
+        try:
+            if not cls.is_reference(ref):
+                raise ValueError('%s is not a valid reference.' % ref)
+
+            return ref.split(cls.separator, 1)[0]
+        except (ValueError, IndexError, AttributeError):
+            raise common_models.InvalidReferenceError(ref=ref)
+
+    @classmethod
+    def get_name(cls, ref):
+        try:
+            if not cls.is_reference(ref):
+                raise ValueError('%s is not a valid reference.' % ref)
+
+            return ref.split(cls.separator, 1)[1]
+        except (ValueError, IndexError, AttributeError):
+            raise common_models.InvalidReferenceError(ref=ref)
+
+    def __repr__(self):
+        return ('<%s resource_type=%s,name=%s,ref=%s>' %
+                (self.__class__.__name__, self.resource_type, self.name, self.ref))
+
+
+class PolicyTypeDB(stormbase.StormBaseDB):
+    """
+    The representation of an PolicyType in the system.
+
+    Attributes:
+        id: See StormBaseAPI
+        name: See StormBaseAPI
+        description: See StormBaseAPI
+        resource_type: The type of resource that this policy type can be applied to.
+        enabled: A flag indicating whether the policies for this type is enabled.
+        module: The python module that implements the policy for this type.
+        parameters: The specification for parameters for the policy type.
+    """
+    ref = me.StringField(required=True)
+    resource_type = me.StringField(
+        required=True,
+        unique_with='name',
+        help_text='The type of resource that this policy type can be applied to.')
+    enabled = me.BooleanField(
+        required=True,
+        default=True,
+        help_text='A flag indicating whether the runner for this type is enabled.')
+    module = me.StringField(
+        required=True,
+        help_text='The python module that implements the policy for this type.')
+    parameters = me.DictField(
+        help_text='The specification for parameters for the policy type.')
+
+    def __init__(self, *args, **kwargs):
+        super(PolicyTypeDB, self).__init__(*args, **kwargs)
+        self.ref = PolicyTypeReference.to_string_reference(resource_type=self.resource_type,
```

Comment:
The fact that this ref is diffrent from other ref in the system bothers me. This ref does not follows the standard `pack.name` convention either.

I see the value in scope by name of the type but couldn't that be simply captured in the name? To deviate from the usual form of ref which is basically the way to uniquely identify, in a human readable form, objects in our system we need absolute conviction. I am not convinced.